src/adapters/mson.es6

```diff
@@ -0,0 +1,201 @@
+      });
+      /* eslint-enable block-scoped-var */
+    }
+  }
+
+  return [renderedContent, objectLike];
+}
+
+/*
+ * Handle the rendering of an element based on its element type. This function
+ * will call itself recursively to handle child elements for objects and
+ * arrays.
+ */
+function handle(name, element, {parent=null, spaces=4, marker='+',
+                                initialMarker='+',
+                                initialIndent=true,
+                                attributesElement=element}) {
+  const elementName = element.element;
+  let str = initialMarker;
+
+  // Start with the item name if it has one.
+  if (name) {
+    str += ` ${name}`;
+  }
+
+  // Next, comes the optional example value
+  if (typeof element.content !== 'object') {
+    if (parent && parent.element !== 'array') {
+      str += ':';
+    }
+    str += ` ${element.content}`;
+  }
+
+  // Then the type and attribute information (e.g. required)
+  let attributes = getTypeAttributes(element, attributesElement.attributes);
+  if (attributes.length) {
+    str += ` (${attributes.join(', ')})`;
+  }
+
+  // This flag determines when to use a block description within a list entry
+  // instead of just ending the list entry line with the description. This
+  // means that some other special values like `+ Properties` will get used
+  // later during rendering.
+  let useLongDescription = false;
+  if (element.attributes &&
+      element.attributes.default !== undefined ||
+      element.attributes.sample !== undefined) {
+    useLongDescription = true;
+  }
+
+  // Finally, an optional description
+  let description = attributesElement.meta.description.toValue();
+  if (description) {
+    if (description.indexOf('\n') !== -1) {
+      // Multiline description, so we can't use the short form!
+      useLongDescription = true;
+    }
+
+    if (useLongDescription) {
+      str += `\n${description}`;
+    } else {
+      str += ` - ${description}`;
+    }
+  }
+
+  // Handle special list items like default/sample here as they are part
+  // of the description, before the content (sub-elements) are rendere.
+  if (element.attributes.default !== undefined) {
+    str += `\n${marker} Default: ${element.attributes.default}\n`;
+  }
+
+  if (element.attributes.sample !== undefined) {
+    str += `\n${marker} Sample: ${element.attributes.sample}\n`;
+  }
+
+  // Now, depending on the content type, we will recursively handle child
+  // elements within objects and arrays.
+  if (element.content && element.content.length) {
+    str += '\n';
+
+    if (elementName === 'dataStructure') {
+      str += '\n';
+    }
+
+    let [renderedContent, objectLike] = handleContent(element, spaces, marker);
+
+    if (!useLongDescription) {
+      str += renderedContent;
+    } else if (renderedContent.length) {
+      // There is rendered content
+      if (objectLike) {
+        str += `\n${marker} Properties\n`;
+      } else {
+        str += `\n${marker} Items\n`;
```

Comment:
Do `Members` ever come into play in addition to `Items` and `Properties`?

https://github.com/apiaryio/mson/blob/master/MSON%20Specification.md#421-member-type-separator